src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
Yes, that is what I understood from the comment that was in the code: the client->proxy half should be able to close both sides if finished/error, but not the other way around.
If we want both sides to be able to close the other one preemptively, then I think is best done with `once`, indeed!